crates/biome_js_analyze/src/analyzers/nursery/no_done_callback.rs

```diff
@@ -0,0 +1,161 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_syntax::{
+    AnyJsArrowFunctionParameters, AnyJsBinding, AnyJsExpression, JsCallExpression, JsParameters,
+    JsTemplateExpression,
+};
+use biome_rowan::{AstNode, TextRange};
+
+declare_rule! {
+    /// Disallow using a callback in asynchronous tests and hooks.
+    ///
+    /// This rule checks the function parameter of hooks & tests for use of the done argument, suggesting you return a promise instead.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// beforeEach(done => {
+    ///     // ...
+    /// });
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// test('myFunction()', done => {
+    ///     // ...
+    /// });
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// beforeEach(async () => {
+    ///     // ...
+    /// });
+    /// ```
+    ///
+    /// ```js
+    /// test('myFunction()', () => {
+    ///     expect(myFunction()).toBeTruthy();
+    /// });
+    /// ```
+    ///
+    pub NoDoneCallback {
+        version: "next",
+        name: "noDoneCallback",
+        recommended: true,
+        source: RuleSource::EslintJest("no-done-callback"),
+        source_kind: RuleSourceKind::Inspired,
```

Comment:
sorry, btw, I just wanna know the reason it's inspired, not same logic?
```suggestion
        source_kind: RuleSourceKind::Inspired,
```

Reply:
You are right, it's the same logic.